### PR TITLE
Update task scheduler using service workers extensibility

### PR DESCRIPTION
**Pull request by maheshkk**
First of all I'm bit new the concept and I plead for your patience with review. 

This merge request :
- Updates task scheduler API spec using service workers extention API.
- Proposing to remove hasPendingMessages() API from the task scheduler (I might be totally missing usecase here) as this feels redundant. Either there will be no pending messages as service worker will always get notified of the task fired, or one can use getPendingTasks to know if any tasks are active and pending system message. 

/cc @jungkees @anssiko

### CODE REVIEW

**Comment by wonsuk73** [comment]
This is covered by #66 merge request.

File: index.html [hunk 1]
```diff
@@ -80,14 +74,15 @@ <h2 class="no-num no-toc" id="editor-draft-—-5-april-2014">
 Editors:
 </dt>
 <dd class="vcard">
-<span class="fn">Christophe Dumez</span>, <span class="org"><a href="http://www.samsung.com/sec">Samsung Electronics,
-Co., Ltd</a></span>, <span class="email"><a href="mailto:ch.dumez@samsung.com">ch.dumez@samsung.com</a></span>
+<span class="fn">Mahesh Kulkarni</span>, <span class="org"><a href="http://www.samsung.com/sec">Samsung Electronics,
+Co., Ltd</a></span>, <span class="email"><a href="mailto:mahesh.kk@samsung.com">mahesh.kk@samsung.com</a></span>
+<span class="fn">Christophe Dumez</span>
```
Comment (kenchris) [inline index.html:79]: isnt he former editor?

File: index.html [hunk 2]
```diff
@@ -96,15 +91,15 @@ <h2 class="no-num no-toc" id="specabstract">
 </h2>
 <p>
 This specification defines an API to schedule a task at a specified time. When the indicated time is reached, the
-application that scheduled the task will be notified via a system message. If the application is not running when this
-happens, the system message will cause the application will be launched. Applications such as an alarm clock or an
+application that scheduled the task will be notified via a system message. System message will be delivered to Service Worker, regardless
```
Comment (kenchris) [inline index.html:94]: Shouldnt we avoid saying "system message" - I have no idea what that is and isnt that what moz used for Firefox OS and in the original spec. This needs rewording which fits with the service worker.

Comment (kenchris) [inline index.html:94]: Maybe look at what the Push API spec says

File: index.html [hunk 3]
```diff
@@ -96,15 +91,15 @@ <h2 class="no-num no-toc" id="specabstract">
...
+of whether the application is active on user agent. Applications such as an alarm clock or an
```
Comment (kenchris) [inline index.html:95]: active on user agent sounds very weird to me. I don't understand what that means

File: index.html [hunk 4]
```diff
@@ -96,15 +91,15 @@ <h2 class="no-num no-toc" id="specabstract">
...
 auto-updater may utilize this API to perform some action at a specified time.
```
Comment (kenchris) [inline index.html:96]: s/some/certain actions at a specific time

File: index.html [hunk 5]
```diff
@@ -153,22 +152,37 @@ <h2 id="introduction"><span class="secno">1 </span>
 <p>
 How to set an alarm 10 minutes from now?
 </p>
-<pre>function onTaskAdded(task) {
-  console.log(task.data.message);
+<pre>// https://example.com/serviceworker.js
+this.ontask = function(task) {
+  alert(task.data.message);
+  console.log("Task scheduled at: " + new Date(task.time));
+  // From here we can write the data to IndexedDB, send it to any open windows,
```
Comment (kenchris) [inline index.html:159]: From here on...

File: index.html [hunk 6]
```diff
@@ -153,22 +152,37 @@ <h2 id="introduction"><span class="secno">1 </span>
...
+  // display a notification, etc.
+}
+
+// https://example.com/webapp.js
```
Comment (kenchris) [inline index.html:163]: W3C is starting to not use real urls for examples and .test instead, like example.test/ I am not sure how this is supposed to be used in specs, @anssiko  ?

File: index.html [hunk 7]
```diff
@@ -153,22 +152,37 @@ <h2 id="introduction"><span class="secno">1 </span>
...
 function onError(error) {
   alert("Sorry, couldn't set the alarm: " + error);
 }
 
-navigator.taskScheduler.add(Date.now() + (10 * 60000), {
-  message: "It's been 10 minutes, your soup is ready!"
-}).then(onTaskAdded, onError);
+navigator.serviceWorker.ready.then(function(serviceWorkerRegistration) {
+  serviceWorkerRegistration.taskScheduler.add(Date.now() + (10 * 60000), {
+    message: "It's been 10 minutes, your soup is ready!"
```
Comment (kenchris) [inline index.html:173]: s/,/; ?

File: index.html [hunk 8]
```diff
@@ -153,22 +152,37 @@ <h2 id="introduction"><span class="secno">1 </span>
...
+});
 </pre>
 <p>
 How to get all the scheduled tasks whose time is in the future?
 </p>
-<pre>navigator.taskScheduler.getPendingTasks().then(function(tasks) {
-  alert("There are " + tasks.length + " tasks set."));
+<pre>navigator.serviceWorker.getRegistration().then(function(registration) {
+  registration.taskScheduler.getPendingTasks().then(function(tasks) {
+    alert("There are " + tasks.length + " tasks set.");
```
Comment (kenchris) [inline index.html:182]: There are .... pending tasks ?

File: index.html [hunk 9]
```diff
@@ -153,22 +152,37 @@ <h2 id="introduction"><span class="secno">1 </span>
...
+  }, function(error) {
+    alert("An error occurred getting the scheduled tasks.");
```
Comment (kenchris) [inline index.html:184]: occurred while ....

File: index.src.html [hunk 10]
```diff
@@ -115,23 +116,38 @@ <h2 id="introduction">
 How to set an alarm 10 minutes from now?
 </p>
 <pre>
+// https://example.com/serviceworker.js
+this.ontask = function(task) {
+  alert(task.data.message);
+  console.log("Task scheduled at: " + new Date(task.time));
+  // From here we can write the data to IndexedDB, send it to any open windows,
```
Comment (kenchris) [inline index.src.html:123]: isnt this duplicated... still "from here on..."

File: index.html (hunk 1 above)
Comment (jungkees) [inline index.html:79]: Already addressed in merge #66.

File: index.html (hunk 2 above)
Comment (jungkees) [inline index.html:94]: #66 addressed most of them and #67 fixes this sentence. However, I'd also expect @maheshkk improve the wordings onwards.

File: index.html (hunk 4 above)
Comment (jungkees) [inline index.html:96]: #67 fixes it.

File: index.html (hunk 5 above)
Comment (jungkees) [inline index.html:159]: #67 fixes it.

File: index.html (hunk 7 above)
Comment (jungkees) [inline index.html:173]: I'm not catching any error here. Am I missing it?

File: index.src.html (hunk 10 above)
Comment (jungkees) [inline index.src.html:123]: It seems that @maheshkk was not accustomed to using anolis at the beginning and edited both .src.html and .html. But from the sequence of his commits, I guess he already got the hang of it.

**Pull request closed without merging**